src/realm/util/terminate.cpp

```diff
@@ -52,10 +58,14 @@ namespace {
 #if REALM_PLATFORM_APPLE
 void nslog(const char* message) noexcept
 {
-    // Standard error goes nowhere for applications managed by launchd, so log to ASL as well.
+    // Standard error goes nowhere for applications managed by launchd,
+    // so log to ASL/unified loggin system logs as well.
     fputs(message, stderr);
+#if REALM_APPLE_OS_LOG
+    os_log_error(OS_LOG_DEFAULT, "%s", message);
```

Comment:
As we discussed on Slack, the format string needs to be `%{public}s`.

From `man os_log`:

> The unified logging system considers dynamic strings and complex dynamic objects to be private, and does not collect them automatically.  To ensure the privacy of users, it is recommended that log messages consist      strictly of static strings and numbers, which are collected automatically by the system.  In situations where it is necessary to capture a dynamic string, and it would not compromise user privacy, you may explic-      itly declare the string public by using the public keyword in the log format string.  For example, %{public}s.  Log arguments can also be specified as private by using the private keyword in the log format string.      For example, %{private}d.